types/firestore.d.ts

```diff
@@ -2377,7 +2463,7 @@ declare namespace FirebaseFirestore {
      * @param documentSnapshot A `DocumentSnapshot` to add.
      * @returns This instance.
      */
-    add<T>(documentSnapshot: DocumentSnapshot<T>): BundleBuilder;
+    add(documentSnapshot: DocumentSnapshot): BundleBuilder;
```

Comment:
add() should be parameterized with `AppDataType` and `DbDataType`. Also on line 2476.